mlflow/tracing/client.py

```diff
@@ -285,27 +285,30 @@ def search_traces(
                 else None
             )
 
-        def download_trace_extra_fields(trace_info: TraceInfo) -> Optional[Trace]:
+        def download_trace_extra_fields(trace_info) -> Optional[Trace]:
```

Comment:
```suggestion
        def download_trace_extra_fields(trace_info: Union[TraceInfo, TraceInfoV3]) -> Optional[Trace]:
```

Let's keep type hint. Actually, can we update the upstream to always pass `TraceInfoV3`?